src/factories/nodeFlowRun.ts

```diff
@@ -44,11 +45,13 @@ export async function flowRunContainerFactory(node: RunGraphNode) {
     resized()
   })
 
-  async function render(): Promise<BoundsContainer> {
+  async function render(node: RunGraphNode): Promise<BoundsContainer> {
+    internalNode = node
+
     await renderBar(node)
-    await renderArrowButton()
-    await renderLabel()
-    await renderBorder()
+    await renderArrowButton(node)
+    await renderLabel(node)
+    await renderBorder(node)
```

Comment:
realized these were all rendering based on the node passed into the factory and not the value passed into the render function.